### ISSUE TITLE
Using ldap3 behind a proxy server

### ISSUE THREAD

**Issue by lourenko** [cannatag/ldap3#453]
Hello, I'm using ldap3 behind a proxy server. My connection looks like this:

```
ldap3.Connection(
... ldap3.Server(
... "ldap://ldap.test.edu",
... allowed_referral_hosts=[("", True)],
... ),
... user="username",
... password="passwd",
... auto_bind=ldap3.AUTO_BIND_NO_TLS,
... raise_exceptions=True,
... )
```
When run, I get this stack trace:

```
LDAPSocketOpenError                       Traceback (most recent call last)
<ipython-input-12-43ac84cd7e35> in <module>()
      7 password="passwd",
      8 auto_bind=ldap3.AUTO_BIND_NO_TLS,
----> 9 raise_exceptions=True,
     10 )

~/.local/lib/python3.5/site-packages/ldap3/core/connection.py in __init__(self, server, user, password, auto_bind, version, authentication, client_strategy, auto_referrals, auto_range, sasl_mechanism, sasl_credentials, check_names, collect_usage, read_only, lazy, raise_exceptions, pool_name, pool_size, pool_lifetime)                          
    274                     if log_enabled(BASIC):
    275                         log(BASIC, 'performing automatic bind for <%s>', self)
--> 276                     self.open(read_server_info=False)
    277                     if self.auto_bind == AUTO_BIND_NO_TLS:
    278                         self.bind(read_server_info=True)

~/.local/lib/python3.5/site-packages/ldap3/strategy/sync.py in open(self, reset_usage, read_server_info)
     52 
     53     def open(self, reset_usage=True, read_server_info=True):
---> 54         BaseStrategy.open(self, reset_usage, read_server_info)
     55         if read_server_info:
     56             try:

~/.local/lib/python3.5/site-packages/ldap3/strategy/base.py in open(self, reset_usage, read_server_info)
    130                         if log_enabled(ERROR):
    131                             log(ERROR, '<%s> for <%s>', exception_history[0][1](exception_history[0][2]), self.connection)
--> 132                         raise exception_history[0][1](exception_history[0][2])
    133                     else:
    134                         if log_enabled(ERROR):

LDAPSocketOpenError: socket connection error: [Errno 110] Connection timed out
```

The same code works fine outside the proxy. If there is any workaround I can use to have authentication for the proxy using ldap3, please let me know.
Thanks!

**Comment by cannatag** [comment]
Hi, you’re getting a socket timeout when trying to open the connection, are you sure you’re able to contact the ldap server through the proxy?

You can try to use telnet to connect to the server on the 389 port and see if you get a connection.

Are you able to connect using another tool (ldapsearch, ldap browser)?

**Comment by lourenko** [comment]
I do not used another tool (ldapsearch, ldap browser), because I just want to try the [django-python3-ldap](https://github.com/etianen/django-python3-ldap) user authentication in Django via LDAP. I tried to run the tests showed [here](https://github.com/etianen/django-python3-ldap/issues/72) in a thread showing similar errors to mine.

**Comment by cannatag** [comment]
I was asking if you were able to perform the operation with another tool just to understand in your connnection is fine! "Connection timeout" let me think that you don't get any response from the proxy, Ket me know if with another tool it works or not.

**Comment by cannatag** [comment]
Any new on this issue? I've just released version 2.5 of ldap3. Can you give a try?

**Comment by lourenko** [comment]
Sorry for the late reply. I'll just try it!

Thank you very much.

Marcos Lourenço

On Sat, Apr 14, 2018 at 12:39 PM, Giovanni Cannata <notifications@github.com
